covid19/models.py

```diff
@@ -135,3 +146,50 @@ def get_total_data(self):
             return [d for d in self.table_data if d['place_type'] == 'state'][0]
         except IndexError:
             return None
+
+    def compare_to_spreadsheet(self, other_spreadsheet):
+        match = lambda e1, e2: e1 and e2 and e1['deaths'] == e2['deaths'] and e1['confirmed'] == e2['confirmed']
+
+        errors = []
+        if not self.date == other_spreadsheet.date:
+            errors.append("Datas das planilhas são diferentes.")
+        if not self.state == other_spreadsheet.state:
+            errors.append("Estados das planilhas são diferentes.")
+
+        if errors:
+            return errors
+
+        for entry in self.table_data:
```

Comment:
Se `other_spreadsheet` (sugiro usar só `other`) tiver dados de municípios diferentes que `self`, esse check não vai pegar.
Sugiro fazer a comparação assim:
- Gera uma property em `StateSpreadsheet` que te devolva um dicionário onde as chaves são o `place_key` e os valores sejam os dados para aquele local, assim esse dicionário contemplará tanto os registros de municípios quanto o total no estado (vou chamar aqui de `self.X` e `other.X`);
- Compara `sorted(self.X.keys())` com `sorted(other.X.keys())` - se forem diferentes, gera N erros:
  - `for err1 in set(self.X.keys()) - set(other.X.keys()): errors.append(f"{err1} está aqui e não lá")`
  - `for err2 in set(other.X.keys()) - set(self.X.keys()): errors.append(f"{err2} está lá e não aqui")`
- Para cada key em `self.X.keys()`, verifica se o valor dessa key é igual a `other.X[key]`;

Reply:
Resolvido em be5d8d7